icssc/AntAlmanac
# Add Custom Events and Schedule Notes to Skeleton Mode

## Summary
1. Custom Events are now viewable in Skeleton Mode. Creation and Editing functionality is disabled.
2. Schedule Notes are now viewable in Skeleton Mode. Creation and Editing functionality is disabled.
3. Styling has been cleaned up for both skeleton and live mode.
4. Schedule Select Popover in the Calendar Toolbar is disabled.

## Before and Afters
### Empty Schedule
Before
<img width="500" alt="Before - Empty Schedule" src="https://github.com/icssc/AntAlmanac/assets/100006999/e2a9da9e-15e2-47f1-bc26-b93ed4fef112">
After
<img width="500" alt="After - Empty Schedule" src="https://github.com/icssc/AntAlmanac/assets/100006999/7a485564-4df3-43af-aec6-1a0c9c873f37">

### Custom Events Box
Before
<img width="500" alt="Before - Custom Events" src="https://github.com/icssc/AntAlmanac/assets/100006999/e759159e-97dc-45fe-97a1-dbbd0f2e6d87">
After
<img width="500" alt="After - Custom Events" src="https://github.com/icssc/AntAlmanac/assets/100006999/72a20e3c-4be4-498e-b4ad-86ef0b5d8783">

### Skeleton Mode
Before
<img width="500" alt="Before - Skeleton Mode" src="https://github.com/icssc/AntAlmanac/assets/100006999/a143066f-617e-43c5-8912-e656ebc314a9">
After
<img width="500" alt="After - Skeleton Mode" src="https://github.com/icssc/AntAlmanac/assets/100006999/a2f785f3-5668-41b2-a6d6-ad9e1094002d">

## Test Plan
1. Load a schedule with courses, custom events, and a note in the *first* schedule (the one at the top of the dropdown)
2. Change `PETERPORTAL_WEBSOC_ENDPOINT` to trigger skeleton mode
3. Check that custom events are there and *are not editable*
4. Check that notes are there and *are not editable*
5. For the Calendar Toolbar, check that the Schedule Dropdown, Finals Button, Undo, Delete, and Add Custom Event buttons are disabled
6. For the App Bar, check that Save/Load, and Export are disabled. Notably, Import remains enabled because refactoring it to subscribe to the AppStore properly is somewhat out of scope. Import will still fail though!
7. For the edge case where there are no course events in any schedule, Skeleton Mode **will not trigger.** This is the current intended behavior until a better "Ping Websoc" option is created. 
8. `CustomEventDialog` was refactored. Test that custom events can still be created as long as day and schedule are checked. Make sure that editing the event from both the Calendar and the Added pane still work (and fills the appropriate data). 

## Issues
Closes #700 and Closes #720

## Future Follow-up
1. Refactor `Import.tsx` and properly subscribe to AppStore for the status of SkeletonMode: #771
2. Utilize react-hook-form to clean up `CustomEventDialog.tsx`
